maven-surefire-common/src/test/java/org/apache/maven/plugin/surefire/AbstractSurefireMojoToolchainsTest.java

```diff
@@ -0,0 +1,107 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.singletonMap;
+import static org.apache.maven.artifact.versioning.VersionRange.createFromVersion;
+import static org.mockito.Mockito.*;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+
+/**
+ * Test for {@link AbstractSurefireMojo}.
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({AbstractSurefireMojo.class, ResolvePathsRequest.class, ReflectionUtils.class})
+@PowerMockIgnore({"org.jacoco.agent.rt.*", "com.vladium.emma.rt.*"})
+public class AbstractSurefireMojoToolchainsTest
+{
+
+    private final AbstractSurefireMojo mojo = new AbstractSurefireMojoTest.Mojo();
+
+    @Before
+    public void setupMojo() throws Exception
+    {
+        MavenSession session = mock( MavenSession.class );
+        mojo.setSession( session );
+
+        MavenExecutionRequest mavenExecutionRequest = mock( MavenExecutionRequest.class );
+        when( session.getRequest() ).thenReturn( mavenExecutionRequest );
+
+        Logger logger = mock( Logger.class );
+        mojo.setLogger( logger );
+
+        final VersionRange surefireVersion = createFromVersion( "1" );
+        Artifact forkedBooter = new DefaultArtifact( "org.apache.maven.surefire",
+            "surefire-booter", surefireVersion, null, "jar", null, mock( ArtifactHandler.class ) );
+
+        mojo.setPluginArtifactMap( singletonMap( "org.apache.maven.surefire:surefire-booter", forkedBooter ) );
+    }
+
+    @Test
+    public void shouldHandleMaven30ToolchainManager() throws Exception
+
+    {
+        mockStatic( ReflectionUtils.class );
+        //TODO: mock this to return a method
+        when( ReflectionUtils.tryGetMethod( ToolchainManager.class, "getToolchains" ) ).thenReturn( null );
+
+        ToolchainManager toolchainManager = mock( ToolchainManager.class );
+
+        AbstractSurefireMojo mojo = spy( this.mojo );
+        when( mojo.getToolchainManager() ).thenReturn( toolchainManager );
+
+        Method method = AbstractSurefireMojo.class.getDeclaredMethod( "setupStuff", null );
+        method.setAccessible( true );
+
+        Map<String, String> toolchainSpec = new HashMap<String, String>();
+        toolchainSpec.put( "version", "1.8" );
+        mojo.setJdkToolchain( toolchainSpec );
+
+        method.invoke( mojo );
+
+        verifyStatic( ResolvePathsRequest.class, times( 1 ) );
+        verify( toolchainManager, times( 1 )).getToolchainFromBuildContext( null, null );
```

Comment:
I know it won't get called, but I also don't know how to get rid of the "Missing method call for verify(mock) "

Reply:
Because you have to call it like [this](https://github.com/apache/maven-surefire/blob/pull/285/maven-surefire-common/src/test/java/org/apache/maven/plugin/surefire/AbstractSurefireMojoTest.java#L372).
Be motivated by the **existing** test [AbstractSurefireMojoTest.java](https://github.com/apache/maven-surefire/blob/pull/285/maven-surefire-common/src/test/java/org/apache/maven/plugin/surefire/AbstractSurefireMojoTest.java). The principles are still the same.